llmfoundry/utils/builders.py

```diff
@@ -90,7 +90,15 @@ def build_callback(name: str, kwargs: Dict[str, Any]) -> Callback:
             'log_optimizer_metrics', True),)
     elif name == 'generate_callback':
         prompts = kwargs.pop('prompts')
-        return Generate(prompts=list(prompts), **kwargs)
+        interval = kwargs.pop('interval', None)
+        # Generate callback used to be batch_log_interval, so this is for backwards compatibility
+        if interval is None:
+            if 'batch_log_interval' in kwargs:
+                interval = f"{kwargs.pop('batch_log_interval')}ba"
```

Comment:
Let's raise a deprecation error here.

Reply:
i assume you mean deprecation warning? but yeah, good call.

Reply:
yup, warning!